packages/optimistic-governor/src/mappings/optimisticGovernor.ts

```diff
@@ -71,3 +95,36 @@ export function handleProposalExecuted(event: ProposalExecuted): void {
 
   proposal.save();
 }
+
+export function handleTargetSet(event: TargetSet): void {
+  let optimisticGovernor = getOrCreateOptimisticGovernor(event.address.toHexString());
+  let safe = getOrCreateSafe(event.params.newTarget.toHexString());
+  safe.optimisticGovernor = event.address.toHexString();
+  optimisticGovernor.safe = event.params.newTarget.toHexString();
+
+  // Check if the Optimistic Governor is enabled on the Safe
+  let safeContract = SafeContract.bind(event.params.newTarget);
+  safe.isOptimisticGovernorEnabled = safeContract.try_isModuleEnabled(event.address).value;
+
+  // Create new Safe data source
+  Safe.create(event.params.newTarget);
+
+  optimisticGovernor.save();
+  safe.save();
+}
+
+export function handleSafeDisabledModule(event: DisabledModule): void {
+  let safe = getOrCreateSafe(event.address.toHexString());
+  if (safe.optimisticGovernor.toLowerCase() == event.params.module.toHexString().toLowerCase()) {
+    safe.isOptimisticGovernorEnabled = false;
+    safe.save();
+  }
+}
+
+export function handleSafeEnabledModule(event: EnabledModule): void {
+  let safe = getOrCreateSafe(event.address.toHexString());
+  if (safe.optimisticGovernor.toLowerCase() == event.params.module.toHexString().toLowerCase()) {
+    safe.isOptimisticGovernorEnabled = true;
```

Comment:
Shouldn't this also check that OG target still points to the same safe?

Reply:
That's already handled here:
https://github.com/UMAprotocol/subgraphs/blob/6fbb7457e7635d9e3931edeea0b7596b04639e77/packages/optimistic-governor/src/mappings/optimisticGovernor.ts#L102